src/AVFoundation/AVCompat.cs

```diff
@@ -10,6 +8,26 @@
 using ObjCRuntime;
 
 namespace AVFoundation {
+#if !XAMCORE_4_0
+	public delegate int AVAudioSourceNodeRenderHandler (bool isSilence, AudioToolbox.AudioTimeStamp timestamp, uint frameCount, ref AudioToolbox.AudioBuffers outputData);
+
+	partial class AVAudioSourceNode {
+		[Obsolete("Use 'AVAudioSourceNode (AVAudioSourceNodeRenderHandler2)' instead.")]
+		public AVAudioSourceNode (AVAudioSourceNodeRenderHandler renderHandler)
+			: base (NSObjectFlag.Empty)
```

Comment:
why did you add `: base (NSObjectFlag.Empty)` here ?

Reply:
this allocates the instance (call `alloc` selector) for something that will throw
it seems you're just running additional code (and allocations) for nothing since you'll throwing right afterward

Reply:
Without `: base (NSObjectFlag.Empty)` the compilation fails with this error:

```
➜  src git:(avaudiosourcenode_fix) ✗ make
GEN      [ios] generated_sources
CSC      [ios/64 bit] Xamarin.iOS.dll
AVFoundation/AVCompat.cs(16,10): error CS1729: 'AVAudioNode' does not contain a constructor that takes 0 arguments
make: *** [build/ios/native-64/Xamarin.iOS.dll] Error 1
```

I looked at the generated code to try and figure out what the correct full (non-binding) constructor signature should look like, which is where the `NSObject...` came from. There are also several other constructors that throw exceptions in `AVCompat.cs` that call their base class constructors (like AVCaptureDataOutputSynchronizer) so it seemed like the way to go!

Is there a hack (or non-hack!) to get around the compilation error? 😄

Reply:
That's because the base type has a `[DisableDefaultCtor]` :(

You should be able to workaround it by adding
```csharp
partial class AVAudioNode {
    internal AVAudioNode () {}
}
```
inside the `#if !XAMCORE_4_0` of `AVCompat.cs`

Reply:
OTOH it's not really a big deal, if you're calling that code you're already in a code path leading to an exception (so perf/memory does not matter much). I was surprised by the addition but, now, I recall it was `Constructor` before which is why it build fine without it earlier :)

Reply:
Ah, got it! The empty constructor does not exist because of `[DisableDefaultCtor]` so we have to call one of the other ones 💡 

Update to manually add an empty constructor to `AVAudioNode` inside of `AVCompat.cs` :)